### ISSUE TITLE
Deduplicate - count ones in binary

### ISSUE THREAD

**Issue by hobovsky** [codewars/content-issues#75]
1. [Count the Ones](https://www.codewars.com/kata/5519e930cd82ff8a9a000216)
* 7 kyu
* The task mentions that conversions to string are restricted, but no translation except JS enforces it.
* Satisfaction 90%, 2k solutions
* 5 pending issues
* 4 languages + 2 pending translations
* Published Apr 2015, author inactive.

2. [Bit Counting](https://www.codewars.com/kata/526571aae218b8ee490006f4) (many languages, high completion, overranked at 6kyu)
* 6 kyu
* Satisfaction 88%, 137k solutions.
* 3 pending issues.
* 24 languages + 2 pending translations.
* Published Oct 2013, author inactive.

inb4 _"rerank 2."_: due to significant amount of solutions, reranking 2. most probably won't happen now.

~~EDIT: after some time, another kata was found by **akar-0**. Please see #89.~~ (retired)

**Comment by JohanWiltink** [comment]
Both have missing random tests in some languages. So both are going to require maintenance.

I have no preference. Well, don't throw away more than a literal ton of solutions I guess.

**Comment by ejini6969** [comment]
Me neither. I'm more prefer to keep the latter since the solutions of that one are a subset of the former (even though the no string requirement is enforced), but since reranking is difficult at this stage, I have no other suggestions on what to do next.

**Comment by Blind4Basics** [comment]
retire 2 and remove restrictions from 1.

Ranking of 1 is more appropriate, and has less languages, so less modifications to do.

**Comment by monadius** [comment]
Retire 1 since it has only 4 languages. This task is quite interesting in languages which are missing in 1 (C, NASM, COBOL, etc.). There should be no restrictions.

**Comment by CiprianAmza** [comment]
I would keep only 2 as there are way more translations and more solutions.

**Comment by Blind4Basics** [comment]
the missing languages could be transfered to the 7kyu, couldn't they?

**Comment by ejini6969** [comment]
The thing is, the first one has restrictions and if it is decided to stay, it can be troublesome to enforce across all languages, and if no restrictions, existing languages have to be modified. So, not just translation migration ~~

**Comment by Blind4Basics** [comment]
> retire 2 **and remove restrictions from 1**.

:whistling:

**Comment by akar-0** [comment]
Like monadius I think it's better and easier to keep 2 and remove 1. Restrictions on 1 are irrelevant IMO, and would be hard or impossible to apply in some languages. It's true a fair rank would be rather 7kyu but I think the benefit doesn't justify the cost.

**Comment by Blind4Basics** [comment]
why do I feel like nobody is reading my suggestions...?

what about removing those damn restrictions in 1??

**Comment by monadius** [comment]
Who is going to move 20+ languages from 2 to 1? I would like to change the retirement process: 1. Move all languages from the kata to be retired first. 2. Retire kata.

**Comment by Blind4Basics** [comment]
but is that a so big problem to lose some or even most of those languages? other users can still add them later if they want. The 6 kyu is beyond recovery about the rank, considering the number of completions. This is our _only shot_ to balance this (edit: well, there is _one_ single other option after that: retiring the 6kyu. But after the 7kyu got reitred!??). Imo, a bit more consistency is better than 20 languages where at most 100 users (overall) were involved in (not sure that sentence is correct... Pretty sure it isn't... 'hope it still conveys the idea).

**Comment by monadius** [comment]
I don't care about the rank at all: there is no big difference between 6 kyu and 7 kyu (especially considering that many old kata are overranked). Available languages are substantially more important. And we are not talking about 1-3 languages here.

**Comment by Blind4Basics** [comment]
fine retire 1.

**Comment by hobovsky** [comment]
> I would like to change the retirement process: 1. Move all languages from the kata to be retired first. 2. Retire kata.

Initially this was my idea, and I still think it would be a better approach. The problem I have with it is that it slows things down. I hoped that having at least a slight impression of progress of the process would motivate others to participate, as users would hopefully notice that efforts are justified. If we were to transfer languages first, and handle duplicates later, they would probably never get handled. On the other hand, when we retire them there's at least some visible result, and follow-up issues (transferring translations, adding tests, etc.) will be potentially picked for processing by someone, or, if there will be no volunteers, by me.

I also hoped, like B4B, that handling duplicate kata would be an opportunity to adjust ranks. Reranking is technically harder than adding translations. OTOH, adding translations, although possible, is time consuming. I kinda hoped for a collective effort here, either in short time perspective (menders and PUs help adding them), or in long term perspective (translations will appear as they usually do). In the end, I might end up doing everything myself, which i'd really like to, as I consider working with translations as the most fun Codewars activity for me. But there's so much to handle that it would take ages :(

So basically I think the issue boils down to following questions:
 - Is deduplication a good opportunity for adjusting ranks? Or maybe deduplication should pick a kata of the best quality, and reranking should be handled separately? It is difficult to rerank kata today, but maybe if we choose a misranked kata to stay, its ranking will be possible to be handled some time in the future?
 - Where is the balance between choosing between a kata of inappropriate rank, but small maintenance effort, and a kata which is ranked accurately, but needs significant effort to be fixed?

I think I will create a separate discussion so we could exchange some ideas.

**Comment by EloiseRosen** [comment]
my vote: removing restriction from # 1, retire 2

**Comment by EloiseRosen** [comment]
And vote against us moving all languages before retiring in general. It's already a slow process, translations will come in over time anyway, and people can add them after if they think it's important.

**Comment by error256** [comment]
Isn't removing _most_ languages even worse than having a duplicate kata?

**Comment by Blind4Basics** [comment]
You're reducing the question to a too much tiny subset of it, there.

_Most_ of these languages are practiced by like... dunno... 0.1 or 0.01% of the users? => one could argue that the duplicates-with-bad-ranking-add-all-necessary-criteria-here affect 99% of the users, while all those languages affect only those 0.1-0.01%. With that PoV, which one is worse?

**Comment by krishsharma0413** [comment]
retire 2 and maintain 1 properly

**Comment by hobovsky** [comment]
Another similar kata was found and reported by **akar-0**. Please tell what you think in #89 .

**Comment by Kacarott** [comment]
I vote to keep 2 for the languages. 20 languages is just not worth losing for a slightly better rank.

**Comment by ejini6969** [comment]
I think after one and a half year, this one can be concluded!
